### PR TITLE
Uses AppID for multi-user installation on IKS

### LINKED ISSUE #2182: Provide options to use IBM Cloud AppID as authentication provider
This is a follow up of #2069 and #2000 

## Today

It uses dex as OIDC auth provider.

## Expected outcome

It provides configuration path to use [IBM Cloud AppID](https://www.ibm.com/cloud/app-id) as OIDC auth provider besides dex.

### PR DESCRIPTION
**Pull request by shawnzhu**
Fixes #2182

Notice this PR depends on https://github.com/kubeflow/manifests/pull/1594

### CODE REVIEW

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 1]
```diff
@@ -183,15 +183,47 @@ kfctl apply -V -f ${CONFIG_URI}
...
 
-The scenario is a GitHub organization owner can authorize its organization members to access a deployed kubeflow. A member of this GitHub organization will be redirected to a page to grant access to the GitHub profile by Kubeflow.
-
-1. Create a new OAuth app in GitHub. Use following setting to register the application:
-    * Homepage URL: `http://localhost:8080/`
-    * Authorization callback URL: `http://localhost:8080/dex/callback`
-1. Once the application is registered, copy and save the `Client ID` and `Client Secret` for use later.
-1. Setup environment variables:
+Run the following steps to deploy Kubeflow with [IBM Cloud AppID](https://cloud.ibm.com/catalog/services/app-id)
+as authentication provider.
```
Comment (animeshsingh) [inline content/en/docs/ibm/deploy/install-kubeflow.md:188]:
```suggestion
as an authentication provider. 
```

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 2]
```diff
@@ -183,15 +183,47 @@ kfctl apply -V -f ${CONFIG_URI}
...
-
-1. Create a new OAuth app in GitHub. Use following setting to register the application:
-    * Homepage URL: `http://localhost:8080/`
-    * Authorization callback URL: `http://localhost:8080/dex/callback`
-1. Once the application is registered, copy and save the `Client ID` and `Client Secret` for use later.
-1. Setup environment variables:
+Run the following steps to deploy Kubeflow with [IBM Cloud AppID](https://cloud.ibm.com/catalog/services/app-id)
+as authentication provider. 
+
+The scenario is a Kubeflow cluster admin configures Kubeflow as an web
```
Comment (animeshsingh) [inline content/en/docs/ibm/deploy/install-kubeflow.md:190]:
```suggestion
The scenario is a Kubeflow cluster admin configures Kubeflow as a web
```

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 3]
```diff
@@ -183,15 +183,47 @@ kfctl apply -V -f ${CONFIG_URI}
...
+application in AppID and manages user authentication with builtin
+identity providers (Cloud Directory, SAML, etc) or custom providers.
```
Comment (animeshsingh) [inline content/en/docs/ibm/deploy/install-kubeflow.md:192]:
```suggestion
identity providers (Cloud Directory, SAML, social log-in with Google or Facebook etc.) or custom providers.
```

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 4]
```diff
@@ -183,15 +183,47 @@ kfctl apply -V -f ${CONFIG_URI}
...
+
+1. Follow the guide [Getting started with App ID](https://cloud.ibm.com/docs/appid?topic=appid-getting-started)
+to create an AppID service instance.
+2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
```
Comment (moficodes) [inline content/en/docs/ibm/deploy/install-kubeflow.md:196]: 2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
to create an application with type _regularwebapp_ under the provisioned AppID

Comment (moficodes) [inline content/en/docs/ibm/deploy/install-kubeflow.md:196]:
```
2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
to create an application with type _reguarwebapp_ under the provioned AppID
```

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 5]
```diff
@@ -183,15 +183,47 @@ kfctl apply -V -f ${CONFIG_URI}
...
+to create an application with type _reguarwebapp_ under the provioned AppID
```
Comment (moficodes) [inline content/en/docs/ibm/deploy/install-kubeflow.md:197]:
```suggestion
2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
to create an application with type _regularwebapp_ under the provisioned AppID
```

**Review by moficodes — CHANGES_REQUESTED** [review]

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 6]
```diff
@@ -183,15 +183,47 @@ kfctl apply -V -f ${CONFIG_URI}
...
+instance. Make sure the _scope_ contains _email_. Then retrieve the following 
+configuration parameters from your AppID:
+    * `clientId`
+    * `secret`
+    * `oAuthServerUrl`
+3. Follow the step [Adding redirect URIs](https://cloud.ibm.com/docs/appid?topic=appid-managing-idp)
+to fill a URL for AppID to redirect to Kubeflow. the URL should look like `https://<kubeflow-FQDN>/login/oidc`. Notice that you could follow the guide [Securing the Kubeflow authentication with HTTPS](../authentication/) for the value of `<kubeflow-FQDN>`.
```
Comment (moficodes) [inline content/en/docs/ibm/deploy/install-kubeflow.md:204]:
```suggestion
to fill a URL for AppID to redirect to Kubeflow. the URL should look like `https://<kubeflow-FQDN>/login/oidc`. Notice that you could follow the guide [Securing the Kubeflow authentication with HTTPS](./authentication/) for the value of `<kubeflow-FQDN>`.
```

Comment (moficodes) [inline content/en/docs/ibm/deploy/install-kubeflow.md:204]: This seem like a circular thing. 

We need the deployment to be complete for to have a istio-ingress ip that gives us the FQDN and to complete the deployment we need the FQDN. 

There needs to be a bit more clarity.

Comment (shawnzhu) [reply content/en/docs/ibm/deploy/install-kubeflow.md]: Moved this step to the very end after kubeflow deployment. see https://github.com/kubeflow/website/pull/2327/commits/a15edf41b62c5fbce380029f8fff7cfe9ad54943

**Review by moficodes — CHANGES_REQUESTED** [review]
just one spelling fix. other that that, looks good.

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 7]
```diff
@@ -183,15 +183,46 @@ kfctl apply -V -f ${CONFIG_URI}
...
+
+The scenario is a Kubeflow cluster admin configures Kubeflow as a web
+application in AppID and manages user authentication with builtin identity
+providers (Cloud Directory, SAML, social log-in with Google or Facebook etc.) or
+custom providers.
+
+1. Follow the guide [Getting started with App ID](https://cloud.ibm.com/docs/appid?topic=appid-getting-started)
+to create an AppID service instance.
+2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
+to create an application with type _regularwebapp_ under the provisoned AppID
```
[the remark this answers is not in the document]
Comment (shawnzhu) [reply content/en/docs/ibm/deploy/install-kubeflow.md]: Thanks, fixed inhttps://github.com/kubeflow/website/pull/2327/commits/a6ffbf3b299e35eba3302e5dd4bee73a7a9ee295

**Review by animeshsingh — COMMENTED** [review]
@shawnzhu we need to merge the manifests repo PR before we merge it. Can we target to get it done today?

**Comment by shawnzhu** [comment]
@animeshsingh the PR for manifests has passed CI: https://github.com/kubeflow/manifests/pull/1594

and I've set this PR as ready for review

**Comment by animeshsingh** [comment]
/lgtm

@moficodes are you good to go?

**Pull request merged**